app/models/operatingsystems/windows.rb

```diff
@@ -9,17 +9,15 @@ def pxe_type
     "waik"
   end
 
-  def pxe_prefix(arch)
-    "boot/windows-#{arch}/".tr(" ", "-")
+  def pxe_prefix(medium_provider)
+    medium_provider.interpolate_vars("boot/windows-$arch-#{medium_provider.unique_id}/").tr(" ", "-")
```

Comment:
Why are you doing the `tr`? Base64 will not contain spaces, remove please.

Reply:
Architecture theoretically can have it.